### PR TITLE
Add `:if` as an alias to `:case` in the expression editor

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/50698
Depends on https://github.com/metabase/metabase/pull/51069

How to verify:
- New -> Question -> Products
- Custom column -> `if([Price] > 50, "Large", "Small")`
- Filter -> Custom expression -> `if([Price] > 50, "Large", "Small") = "Large"`
- Aggregation -> Custom expression -> `sum(if([Price] > 50, 1, 0))`
- Preview each step and make sure the results make sense

Note that there is a separate bug with expression matching in the filter widget atm https://github.com/metabase/metabase/pull/51089. E2E tests to be added separately.

### CODE REVIEW

File: frontend/src/metabase-lib/v1/expressions/helper-text-strings.ts [hunk 1]
```diff
@@ -866,6 +866,31 @@ const HELPER_TEXT_STRINGS: HelpTextConfig[] = [
     ],
     docsPage: "case",
   },
+  {
+    name: "if",
+    structure: "if",
+    description: () =>
+      t`Tests an expression against a list of cases and returns the corresponding value of the first matching case, with an optional default value if nothing else is met.`,
```
Comment (mngr) [inline frontend/src/metabase-lib/v1/expressions/helper-text-strings.ts:873]:
```suggestion
      t`Alias for case(). Tests an expression against a list of cases and returns the corresponding value of the first matching case, with an optional default value if nothing else is met.`,
```

**Review by mngr — CHANGES_REQUESTED** [review]
Let's add `Alias for case(). ` text to the description of if() and `Alias for if(). ` text to the description of case()

**Comment by ranquild** [comment]
@mngr updated description

File: frontend/src/metabase-lib/v1/expressions/recursive-parser.js [hunk 2]
```diff
@@ -357,7 +358,7 @@ export const adjustMultiArgOptions = tree =>
 export const adjustBooleans = tree =>
   modify(tree, node => {
     if (Array.isArray(node)) {
-      if (node?.[0] === "case") {
+      if (isCaseOrIfOperator(node[0])) {
```
Comment (romeovs) [inline frontend/src/metabase-lib/v1/expressions/recursive-parser.js:361]: Not sure if it was an intentional change but the new code does not accommodate for cases where `node` is nullish.

**Review by romeovs — APPROVED** [review]
Just one issue in the comments

**Review by mngr — APPROVED** [review]

File: frontend/src/metabase-lib/v1/expressions/recursive-parser.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase-lib/v1/expressions/recursive-parser.js:361]: This is within `Array.isArray` block so `node`  should not be `null` or `undefined`

**Pull request merged**